src/libAtomVM/nifs.c

```diff
@@ -2809,11 +2809,35 @@ static term nif_erlang_binary_to_term(Context *ctx, int argc, term argv[])
 
 static term nif_erlang_term_to_binary(Context *ctx, int argc, term argv[])
 {
-    if (argc != 1) {
-        RAISE_ERROR(BADARG_ATOM);
+    bool allow_latin1_encoding = false;
+    if (argc == 2) {
+        term options = argv[1];
+        VALIDATE_VALUE(options, term_is_list);
+        int proper = 0;
+        if (term_list_length(options, &proper) == 0) {
```

Comment:
I think we should rather use `interop_proplist_get_value` function instead of implementing this logic from scratch.

Reply:
Good point.  Fixed.